plugin/plugin.go

```diff
@@ -154,9 +160,23 @@ func (p *plugin) generateRegexVars(file *generator.FileDescriptor, message *gene
 	ccTypeName := generator.CamelCaseSlice(message.TypeName())
 	for _, field := range message.Field {
 		validator := getFieldValidatorIfAny(field)
-		if validator != nil && validator.Regex != nil {
+		if validator != nil {
 			fieldName := p.GetOneOfFieldName(message, field)
-			p.P(`var `, p.regexName(ccTypeName, fieldName), ` = `, p.regexPkg.Use(), `.MustCompile(`, "`", *validator.Regex, "`", `)`)
+
+			if validator.Regex != nil && validator.UuidVer != nil {
+				fmt.Fprintf(os.Stderr, "WARNING: regex and uuid validator is set for field %v.%v, only one of them can be set, regex validator will be ignored", ccTypeName, fieldName)
+			}
+			if validator.UuidVer != nil {
+				uuid, err := getUUIDRegex(validator.UuidVer)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "WARNING: field %v.%v error %s.\n", ccTypeName, fieldName, err)
+					continue
+				}
+				validator.Regex = &uuid
+			}
+			if validator.Regex != nil {
```

Comment:
Nit: make this chain into an `else if` one to clearly state it as a disjunction rather than a succession of tests any of which might match.

Reply:
I'm not sure If I understand the idea.
Do you mean to change it to this?:

```go
  func (p *plugin) generateRegexVars(file *generator.FileDescriptor, message *generator.Descriptor) {
          ccTypeName := generator.CamelCaseSlice(message.TypeName())
          for _, field := range message.Field {
                  validator := getFieldValidatorIfAny(field)
                  if validator != nil {
                          fieldName := p.GetOneOfFieldName(message, field)
  
                          if validator.Regex != nil && validator.UuidVer != nil {
                                  fmt.Fprintf(os.Stderr, "WARNING: regex and uuid validator is set for field %v.%v, only one of them can be set, regex validator will be ignored", ccTypeName, fieldName)
                          } else if validator.UuidVer != nil {
                                  uuid, err := getUUIDRegex(validator.UuidVer)
                                  if err != nil {
                                          fmt.Fprintf(os.Stderr, "WARNING: field %v.%v error %s.\n", ccTypeName, fieldName, err)
                                  } else {
                                          validator.Regex = &uuid
                                  }
                          } else if validator.Regex != nil {
                                  p.P(`var `, p.regexName(ccTypeName, fieldName), ` = `, p.regexPkg.Use(), `.MustCompile(`, "`", *validator.Regex, "`", `)`)
                          }
  
                          if validator.Regex != nil {
                                  p.P(`var `, p.regexName(ccTypeName, fieldName), ` = `, p.regexPkg.Use(), `.MustCompile(`, "`", *validator.Regex, "`", `)`)
                          }
                  }                                                                                                                                                                                                                                            
          }
  }
```

Reply:
Also fold in the first `if` 🙂 
```go
validator := getFieldValidatorIfAny(field)
if validator != nil {
	fieldName := p.GetOneOfFieldName(message, field)
	if validator.Regex != nil && validator.UuidVer != nil {
		fmt.Fprintf(os.Stderr, "WARNING: regex and uuid validator is set for field %v.%v, only one of them can be set, regex validator will be ignored", ccTypeName, fieldName)
	} else if validator.UuidVer != nil {
		uuid, err := getUUIDRegex(validator.UuidVer)
		if err != nil {
			fmt.Fprintf(os.Stderr, "WARNING: field %v.%v error %s.\n", ccTypeName, fieldName, err)
		} else {
			validator.Regex = &uuid
			p.P(`var `, p.regexName(ccTypeName, fieldName), ` = `, p.regexPkg.Use(), `.MustCompile(`, "`", *validator.Regex, "`", `)`)
		}                         
	} else if validator.Regex != nil {
		p.P(`var `, p.regexName(ccTypeName, fieldName), ` = `, p.regexPkg.Use(), `.MustCompile(`, "`", *validator.Regex, "`", `)`)
	}
}
```

Reply:
This would change the logic.
If `UuidVer` and `Regex` would be set a warning would be printed and none of them would be set.
Currently if both are set a warning is printed and the Uuid validation takes effect.

Should I change to the variant that you proposed + adapt the warning message?

Reply:
That would be great indeed. In the longer run I'd move this to be an actual error that is properly reported and fails a build. But this is something that will need to wait a wider rewrite of that logic.

Reply:
I changed according to your suggestion